RecoCTPPS/ProtonReconstruction/plugins/CTPPSProtonProducer.cc

```diff
@@ -150,60 +221,230 @@ void CTPPSProtonProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSe
+          const auto &i = idx_pairs[pr_idx].first;
+          const auto &j = idx_pairs[pr_idx].second;
+
+          // skip non-unique associations
+          if (idx_pair_multiplicity[i] > 1 || idx_pair_multiplicity[j] > 1)
+            continue;
+
+          const auto &tr_i = hTracks->at(i);
+          const auto &tr_j = hTracks->at(j);
+
+          const double z_i = hGeometry->getRPTranslation(tr_i.getRPId()).z();
+          const double z_j = hGeometry->getRPTranslation(tr_j.getRPId()).z();
+
+          for (const auto &ti : timingSelection[arm_it.first])
+          {
+            const auto &tr_ti = hTracks->at(ti);
+
+            // skip if timing RP saturated (high track multiplicity)
+            if (timing_RP_track_multiplicity[tr_ti.getRPId()] > max_n_timing_tracks_)
+              continue;
+
+            // interpolation from tracking RPs
+            const double z_ti = - hGeometry->getRPTranslation(tr_ti.getRPId()).z(); // the minus sign fixes a bug in the diamond geometry
+            const double f_i = (z_ti - z_j) / (z_i - z_j), f_j = (z_i - z_ti) / (z_i - z_j);
+            const double x_inter = f_i * tr_i.getX() + f_j * tr_j.getX();
+            const double x_inter_unc_sq = f_i*f_i * tr_i.getXUnc()*tr_i.getXUnc() + f_j*f_j * tr_j.getXUnc()*tr_j.getXUnc();
+
+            const double de_x = tr_ti.getX() - x_inter;
+            const double de_x_unc = sqrt(tr_ti.getXUnc()*tr_ti.getXUnc() + x_inter_unc_sq);
+
+            const bool matching = (fabs(de_x) <= de_x_unc);
+
+            ssLog << "ti=" << ti << ", i=" << i << ", j=" << j
+              << " | z_ti=" << z_ti << ", z_i=" << z_i << ", z_j=" << z_j
+              << " | x_ti=" << tr_ti.getX() << ", x_inter=" << x_inter << ", de_x=" << de_x << ", de_x_unc=" << de_x_unc
+              << ", matching=" << matching << std::endl;
+
+            if (!matching)
+              continue;
+
+            mached_timing_track_indeces[pr_idx].push_back(ti);
+            matched_timing_track_multiplicity[ti]++;
+          }
+        }
+
+        // process associated tracks
+        for (unsigned int pr_idx = 0; pr_idx < idx_pairs.size(); ++pr_idx)
+        {
+          const auto &i = idx_pairs[pr_idx].first;
+          const auto &j = idx_pairs[pr_idx].second;
+
+          // skip non-unique associations of tracking-RP tracks
+          if (idx_pair_multiplicity[i] > 1 || idx_pair_multiplicity[j] > 1)
+            continue;
+
+          if (verbosity_)
+            ssLog << std::endl << "* reconstruction from tracking-RP tracks: " << i << ", " << j << " and timing-RP tracks: ";
+
+          // process tracking-RP data
+          CTPPSLocalTrackLiteRefVector sel_tracks;
+          sel_tracks.push_back(CTPPSLocalTrackLiteRef(hTracks, i));
+          sel_tracks.push_back(CTPPSLocalTrackLiteRef(hTracks, j));
+          reco::ForwardProton proton = algorithm_.reconstructFromMultiRP(sel_tracks, *hLHCInfo, ssLog);
+
+          // process timing-RP data
+          double Sw=0., Swt=0.;
```

Comment:
start local variables and data members with lower case letter. Capitalization is reserved for class names.